cmd/kops-controller/pkg/clusterapi/builder.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterapi
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/pkg/apis/kops/registry"
+	kopsv1alpha2 "k8s.io/kops/pkg/apis/kops/v1alpha2"
+	"k8s.io/kops/pkg/client/simple"
+	"k8s.io/kops/pkg/kopscodecs"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup"
+)
+
+type Builder struct {
+	ConfigServer *cloudup.ConfigServer
+	Clientset    simple.Clientset
+}
+
+// ClusterAPIBuilder is implemented by the model task that knows how to map an InstanceGroup to k8s objects
+type ClusterAPIBuilder interface {
+	MapToClusterAPI(cluster *kops.Cluster, ig *kops.InstanceGroup) ([]*unstructured.Unstructured, error)
+}
+
+func (b *Builder) BuildMachineDeployment(clusterObj *kopsv1alpha2.Cluster, igObj *kopsv1alpha2.InstanceGroup) ([]*unstructured.Unstructured, error) {
+	cloudup.AlphaAllowGCE.SetEnabled(true)
```

Comment:
I guess it is necessary to enable this builder to run GCE. But I wonder if we can move it somewhere else, where it does not bloat a unrelated context.